test/functional/change_stream.test.js

```diff
@@ -874,7 +874,7 @@ describe('Change Streams', function () {
           .next()
           .then(function () {
             // We should never execute this line because calling changeStream.next() should throw an error
-            throw new Error(
+            throw new MongoChangeStreamError(
```

Comment:
This should be left as a plain Error, no?

Reply:
Are we not changing errors in tests too? Happy to make the change, but not too clear on why 😅

Reply:
[related](https://github.com/mongodb/node-mongodb-native/pull/2912#discussion_r682825112)
The usage of errors here is to fail the test when an unreachable spot is reached. They aren't being used to test "does the driver return such and such error in some given state" so there isn't a reason to change these. But also it is a little confusing to make them a specific error type because it implies there is a purpose for this error but when reading the test there isn't code that will catch and handle the error.

Reply:
The errors thrown by the tests are only for use by the tests themselves, they are the equivalent of an `expect.fail`. The errors here will never be user facing, so there is no need to specially type them, since they aren't coming from the driver, but rather from our own test setup.

The only time you have to update the types of the errors in the tests is when we are *checking* an error that *is* thrown by the driver itself.

Reply:
Ahhh, makes sense. That's a lot clearer now, thank you guys! Will keep this in mind for future error PRs :)